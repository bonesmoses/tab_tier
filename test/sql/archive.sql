\set ECHO none

BEGIN;
\set ECHO all

CREATE EXTENSION tab_tier;

CREATE SCHEMA tiertest;

SET search_path TO tiertest;

CREATE TABLE tier_test (foo INT, dt TIMESTAMP WITH TIME ZONE);

INSERT INTO tier_test
SELECT a.id, '2016-12-15'::DATE - (a.id::TEXT || 'd')::INTERVAL
  FROM generate_series(1, 200) a (id);

UPDATE tab_tier.tier_part
   SET is_archived = TRUE
 WHERE check_start < '2016-12-15'::DATE - INTERVAL '5 months';

SELECT tab_tier.drop_archived_tiers();

SELECT count(*)
  FROM tab_tier.tier_part
 WHERE part_schema = 'tiertest';

SELECT count(*)
  FROM pg_tables
 WHERE schemaname = 'tiertest'
   AND tablename LIKE '%\_part\_%';

ROLLBACK;
