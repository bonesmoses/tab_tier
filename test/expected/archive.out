\set ECHO none
CREATE EXTENSION tab_tier;
CREATE SCHEMA tiertest;
SET search_path TO tiertest;
CREATE TABLE tier_test (foo INT, dt TIMESTAMP WITH TIME ZONE);
INSERT INTO tier_test
SELECT a.id, '2016-12-15'::DATE - (a.id::TEXT || 'd')::INTERVAL
  FROM generate_series(1, 200) a (id);
UPDATE tab_tier.tier_part
   SET is_archived = TRUE
 WHERE check_start < '2016-12-15'::DATE - INTERVAL '5 months';
SELECT tab_tier.drop_archived_tiers();
 drop_archived_tiers 
---------------------
 
(1 row)

SELECT count(*)
  FROM tab_tier.tier_part
 WHERE part_schema = 'tiertest';
 count 
-------
     0
(1 row)

SELECT count(*)
  FROM pg_tables
 WHERE schemaname = 'tiertest'
   AND tablename LIKE '%\_part\_%';
 count 
-------
     0
(1 row)

ROLLBACK;
